apm-lambda-extension/main.go

```diff
@@ -131,15 +132,24 @@ func main() {
 			backgroundDataSendWg.Add(1)
 			go func() {
 				defer backgroundDataSendWg.Done()
+				if !extension.IsTransportStatusHealthy() {
+					return
+				}
 				for {
 					select {
 					case <-funcDone:
 						log.Println("funcDone signal received, not processing any more agent data")
 						return
 					case agentData := <-agentDataChannel:
+						if !extension.IsTransportStatusHealthy() {
+							extension.EnqueueAPMData(agentDataChannel, agentData)
+							return
+						}
```

Comment:
Seems redundant as if the transport status is not healthy, an error gets returned and the agent data is put back to the channel, too.

Reply:
Solved in 73a25ff